lib/ret/avatar.ex

```diff
@@ -103,14 +108,18 @@ defmodule Ret.Avatar do
     |> unique_constraint(:avatar_sid)
     |> put_assoc(:account, account)
     |> put_assoc(:parent_avatar, parent_avatar)
-    |> put_owned_files(owned_files)
+    |> put_owned_files(owned_files_map)
     |> AvatarSlug.maybe_generate_slug()
     |> AvatarSlug.unique_constraint()
   end
 
-  defp put_owned_files(changeset, owned_files) do
-    Enum.reduce(owned_files, changeset, fn {key, file}, changes ->
-      changes |> put_change(:"#{key}_owned_file_id", file.owned_file_id)
+  defp put_owned_files(changeset, owned_files_map) do
+    Enum.reduce(owned_files_map, changeset, fn
+      {key, :remove}, changes ->
+        changes |> put_assoc(:"#{key}_owned_file", nil)
```

Comment:
i'd probably extract the string expression into a var just to DRY it up a bit

Reply:
I could break it into another function but that seems overkill, unless there is a way to define some intermediate stuff in a functions with multiple clauses like this.